tests/unit/src/main/scala/tests/TestingServer.scala

```diff
@@ -1213,42 +1224,43 @@ final class TestingServer(
     val identifier = path.toTextDocumentIdentifier
     val occurrences = ListBuffer.empty[s.SymbolOccurrence]
     var last = List[String]()
-    Trees.defaultTokenizerDialect(input).tokenize.get.foreach { token =>
-      val params = token.toPositionParams(identifier)
-      val definition = server
-        .definitionOrReferences(params, definitionOnly = true)
-        .asJava
-        .get()
-      definition.definition.foreach { path =>
-        if (path.isDependencySource(workspace)) {
-          readonlySources(path.toNIO.getFileName.toString) = path
+    trees.getDialect(AbsolutePath(input.path))(input).tokenize.get.foreach {
```

Comment:
We could add something like:

```suggestion
    trees.tokenize(AbsolutePath(input.path)).foreach {
```
tokenize would return Option[Array[Token]]

Reply:
Ah good idea.

Reply:
Actually, looking at this I can't this to work right. Where are the implicit tokenize and dialect coming from?

Reply:
I think `scala.meta._`

Reply:
🤔 I see we use it in `InferredTypeProvider` and indeed that works on `<string>.tokenize`, but it doesn't seem to work here on trees unless I'm totally doing something wrong.

Reply:
Ach no, I just meant we need to add that method to Trees. Sorry if I was not clear enough.

Reply:
😆 honestly, that's what I first thought, when trying to do it I did a goto definition on `tokenize` where I was using it, and it took me to the scala meta tokenize extension method. Then I thought "ooooo, he must be trying to tell me to use this".... communication 😆

Reply:
Yeah, I forgot there is so much magic with those extension methods.

Reply:
haha, alright, fixed!